nvm.sh

```diff
@@ -2519,6 +2519,41 @@ nvm() {
         esac
       done
 
+      ADDITIONAL_PARAMETERS=''
+      local PROVIDED_REINSTALL_PACKAGES_FROM
+      local REINSTALL_PACKAGES_FROM
+      local SKIP_DEFAULT_PACKAGES
+      local DEFAULT_PACKAGES
+
+      while [ $# -ne 0 ]
+      do
+        case "$1" in
+          --reinstall-packages-from=*)
+            PROVIDED_REINSTALL_PACKAGES_FROM="$(nvm_echo "$1" | command cut -c 27-)"
+            if [ -z "${PROVIDED_REINSTALL_PACKAGES_FROM}" ]; then
+              nvm_err 'If --reinstall-packages-from is provided, it must point to an installed version of node.'
+              return 6
+            fi
+            REINSTALL_PACKAGES_FROM="$(nvm_version "$PROVIDED_REINSTALL_PACKAGES_FROM")" ||:
+          ;;
+          --reinstall-packages-from)
+            nvm_err 'If --reinstall-packages-from is provided, it must point to an installed version of node using `=`.'
+            return 6
+          ;;
+          --copy-packages-from=*)
+            PROVIDED_REINSTALL_PACKAGES_FROM="$(nvm_echo "$1" | command cut -c 22-)"
+            REINSTALL_PACKAGES_FROM="$(nvm_version "$PROVIDED_REINSTALL_PACKAGES_FROM")" ||:
+          ;;
+          --skip-default-packages)
+            SKIP_DEFAULT_PACKAGES=true
+          ;;
+          *)
+            ADDITIONAL_PARAMETERS="$ADDITIONAL_PARAMETERS $1"
+          ;;
+        esac
+        shift
```

Comment:
this `shift` will change what `$1` is in the next few lines; i think it needs to be extracted first, otherwise it will always end up in `ADDITIONAL_PARAMETERS`

Reply:
This shift is fine. The param only ends up in `ADDITIONAL_PARAMETERS` if it didn't match the above cases. I believe `ADDITIONAL_PARAMETERS` gathers all the unknown params and passes them to the `nvm_install_source` function for use there to customize the binary compile.

However that does present a problem, since my fix is to make the version specifier optional when `--lts` is used. But it wouldn't be able to be differentiated from these also optional compile params. 

I forgot about the optional compile params. Basically the `install` command is fucked unless we are willing to break the current way people are using it. As I see it we can't cleanly fix the --lts use case where no version is given without breaking params after it. Closing PR now.

Reply:
Let's not be so hasty; I'm sure there's some way it can be addressed, it just might not be super clean.